nomad/state/state_store.go

```diff
@@ -2126,20 +2126,20 @@ func (s *StateStore) CSIVolumes(ws memdb.WatchSet) (memdb.ResultIterator, error)
 func (s *StateStore) CSIVolumeByID(ws memdb.WatchSet, namespace, id string) (*structs.CSIVolume, error) {
 	txn := s.db.ReadTxn()
 
-	watchCh, obj, err := txn.FirstWatch("csi_volumes", "id_prefix", namespace, id)
+	obj, err := txn.First("csi_volumes", "id", namespace, id)
 	if err != nil {
-		return nil, fmt.Errorf("volume lookup failed: %s %v", id, err)
+		return nil, fmt.Errorf("volume lookup failed: %s: %v", id, err)
 	}
-
-	ws.Add(watchCh)
-
 	if obj == nil {
 		return nil, nil
 	}
+	vol, ok := obj.(*structs.CSIVolume)
+	if !ok {
+		return nil, fmt.Errorf("volume row conversion error")
```

Comment:
This seems like a violated invariant, potentially a data corruption issue.  Not sure what should happen here.  It may be nice to log and include the `obj` type as well?

Reply:
It looks like the rest of the state store code actually just tosses this conversion error out and lets us panic later. That's probably the right move.

Reply:
Done in 4ae9cb8